src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -34,20 +45,60 @@ public DeleteCommand(Index targetIndex) {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAY_INDEX);
-        }
+        ContextType context = model.getContext().getType();
 
-        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
-        if (model.getActivityBook().hasPerson(personToDelete.getPrimaryKey())) {
-            throw new CommandException(String.format(MESSAGE_PERSON_INVOLVED_ACTIVITY,
-                            personToDelete.getName()));
-        }
+        switch (context) {
+        case LIST_CONTACT: // delete contact
+            List<Person> lastShownList = model.getFilteredPersonList();
+
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+            }
+
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            if (model.getActivityBook().hasPerson(personToDelete.getPrimaryKey())) {
+                throw new CommandException(String.format(MESSAGE_PERSON_INVOLVED_ACTIVITY, personToDelete.getName()));
+            }
+
+            Context listContact = new Context(ContextType.LIST_CONTACT);
+            model.deletePerson(personToDelete);
+            model.setContext(listContact);
 
-        model.deletePerson(personToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+            return new CommandResult(String.format(MESSAGE_DELETION_SUCCESS, personToDelete.getName().fullName),
+                    listContact);
+            
+        case LIST_ACTIVITY: // delete activity
+            List<Activity> activityList = model.getFilteredActivityList();
+
+            if (targetIndex.getZeroBased() >= activityList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+            }
+
+            Activity activityToDelete = activityList.get(targetIndex.getZeroBased());
+            Context listActivity = new Context(ContextType.LIST_ACTIVITY);
+            model.deleteActivity(activityToDelete);
+            model.setContext(listActivity);
+
+            return new CommandResult(String.format(MESSAGE_DELETION_SUCCESS, activityToDelete.getTitle().title),
+                    listActivity);
+
+        case VIEW_ACTIVITY: // delete expense in an activity
+            Activity activity = model.getContext().getActivity().get();
+            if (targetIndex.getZeroBased() >= activity.getExpenses().size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+            }
+
+            Expense expense = activity.getExpenses().get(targetIndex.getZeroBased()); // soft deletes the expense
+            expense.delete();
+            Context thisActivity = new Context(activity);
```

Comment:
This constructor for `Context` is okay since it is understood that if a context involves a specific activity, that the active context should be `VIEW_ACTIVITY`.